packages/forklift-console-plugin/src/modules/Plans/views/details/tabs/VirtualMachines/modals/PlanVMsDeleteModal.tsx

```diff
@@ -23,6 +24,15 @@ export const PlanVMsDeleteModal: React.FC<PlanVMsDeleteModalProps> = ({ plan, se
   const vms = (plan?.spec?.vms || []).filter((vm) => !selected.includes(vm.id)) || [];
 
   React.useEffect(() => {
+    if (isPlanArchived(plan)) {
+      setAlertMessage(
```

Comment:
why setting the whole comopnent as a state? make it much harder for React to compare, and not only the msg string? also, you will reduce the reputation in the code here of 

`<AlertMessageForModals
          title={t('Error')}
          message={something}
        />,`

instead of declaring it twice, only once..

Reply:
in current line 100 u can do after it
`{msg && <AlertMessageForModals title={t('Error')} message={something} />}`